src/DeepL.php

```diff
@@ -281,194 +342,34 @@ protected function checkLanguages($sourceLanguage, $destinationLanguage)
-        array $splittingTags = null
+    protected function buildQuery(
+        $paramsArray
     ) {
-        $url = $this->buildBaseUrl($resource);
-
-        if (false === empty($sourceLanguage)) {
-            $url .= '&'.sprintf(self::API_URL_SOURCE_LANG, strtolower($sourceLanguage));
-        }
-
-        if (false === empty($destinationLanguage)) {
-            $url .= '&'.sprintf(self::API_URL_DESTINATION_LANG, strtolower($destinationLanguage));
-        }
-
-        if (false === empty($tagHandling)) {
-            $url .= '&'.sprintf(self::API_URL_TAG_HANDLING, $tagHandling);
-        }
-
-        if (false === empty($ignoreTags)) {
-            $url .= '&'.sprintf(self::API_URL_IGNORE_TAGS, urlencode(implode(',', $ignoreTags)));
-        }
-
-        if (false === empty($formality)) {
-            $url .= '&'.sprintf(self::API_URL_FORMALITY, $formality);
-        }
-
-        if (false === empty($splitSentences)) {
-            $url .= '&'.sprintf(self::API_URL_SPLIT_SENTENCES, $splitSentences);
-        }
-
-        if (false === empty($preserveFormatting)) {
-            $url .= '&'.sprintf(self::API_URL_PRESERVE_FORMATTING, $preserveFormatting);
-        }
-
-        if (false === empty($nonSplittingTags)) {
-            $url .= '&'.sprintf(self::API_URL_NON_SPLITTING_TAGS, urlencode(implode(',', $nonSplittingTags)));
-        }
-
-        if (0 === $outlineDetection) {
-            $url .= '&'.sprintf(self::API_URL_OUTLINE_DETECTION, $outlineDetection);
-        }
-
-        if (false === empty($splittingTags)) {
-            $url .= '&'.sprintf(self::API_URL_SPLITTING_TAGS, urlencode(implode(',', $splittingTags)));
-        }
-
-        return $url;
-    }
-
-    /**
-     * Creates the Base-Url which all of the 3 API-recourses have in common.
-     *
-     * @param string $resource
-     *
-     * @return string
-     */
-    protected function buildBaseUrl($resource = 'translate')
-    {
-        $url = sprintf(
-            self::API_URL_BASE,
-            self::API_URL_SCHEMA,
-            $this->host,
-            $this->apiVersion,
-            $resource,
-            $this->authKey
-        );
-
-        return $url;
-    }
-
-    /**
-     * Build the body for the DeepL API request
-     *
-     * @param string|string[] $text
-     *
-     * @return string
-     */
-    protected function buildBody($text)
-    {
-        $body  = '';
-        $first = true;
-
-        if (!is_array($text)) {
-            $text = (array)$text;
-        }
-
-        foreach ($text as $textElement) {
-            $body .= ($first ? '' : '&').sprintf(self::API_URL_TEXT, rawurlencode($textElement));
-
-            if ($first) {
-                $first = false;
+        if (true === is_array($paramsArray['text'])) {
+            $text = $paramsArray['text'];
+            unset($paramsArray['text']);
+            $textString = '';
+            foreach ($text as $textElement) {
+                $textString .= '&text='.rawurlencode($textElement);
```

Comment:
```suggestion
            $textString = '';
            foreach ($paramsArray['text'] as $textElement) {
                $textString .= '&text='.rawurlencode($textElement);
            }
            $paramsArray['text'] = $textString;
```
the if clause at the end would be unnecessary also.

Reply:
That would lead for the textstring to get double encoded and mixed up:
&text=text&text=more%20text&text=even%20more%20text
would become:
text=%26text%3Dtext%26text%3Dmore%2520text%26text%3Deven%2520more%2520text

Reply:
Your're right. I didn't consider the http_build_query function. My fault.